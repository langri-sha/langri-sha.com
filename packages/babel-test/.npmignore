# ~~ Generated by projen. To modify, edit .projenrc.mts and run "npx projen".
.*
*.test.*
__snapshots__/
tsconfig*.json
fixtures/
