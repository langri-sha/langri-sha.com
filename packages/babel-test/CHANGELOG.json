{
  "name": "@langri-sha/babel-test",
  "entries": [
    {
      "date": "Wed, 17 Jul 2024 10:48:16 GMT",
      "version": "0.6.1",
      "tag": "@langri-sha/babel-test_v0.6.1",
      "comments": {
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "fix(workspace): Use correct GitHub directory"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.11"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "feat(root): Migrate to ESM"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "feat(eslint-config): Add `import/order` rule"
          }
        ]
      }
    },
    {
      "date": "Mon, 08 Jul 2024 21:43:05 GMT",
      "version": "0.6.0",
      "tag": "@langri-sha/babel-test_v0.6.0",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.5"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update node.js to v20.15.0"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use correct output directory"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.6"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/ramda to v0.30.1"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.3"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.10"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.7"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.8"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Add explicit return types"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(tsconfig): Use `${configDir}`"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Add repository path"
          }
        ],
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Use `vitest`"
          }
        ]
      }
    },
    {
      "date": "Sat, 15 Jun 2024 09:09:06 GMT",
      "version": "0.5.0",
      "tag": "@langri-sha/babel-test_v0.5.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "7ac78ea2d172ae9e30f23d213752297fb8436a88",
            "comment": "feat(babel-preset): Hoist TS types to peers"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "7ac78ea2d172ae9e30f23d213752297fb8436a88",
            "comment": "feat(babel-test): Manage package with projen"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "7ac78ea2d172ae9e30f23d213752297fb8436a88",
            "comment": "feat(projen): Reconfigure project expressions"
          }
        ],
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "7ac78ea2d172ae9e30f23d213752297fb8436a88",
            "comment": "chore(deps): chore(deps): update node.js to v20.14.0"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "7ac78ea2d172ae9e30f23d213752297fb8436a88",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.14.2"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "7ac78ea2d172ae9e30f23d213752297fb8436a88",
            "comment": "chore(deps): fix(deps): update dependency ramda to v0.30.1"
          }
        ]
      }
    },
    {
      "date": "Tue, 21 May 2024 05:22:50 GMT",
      "version": "0.4.0",
      "tag": "@langri-sha/babel-test_v0.4.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "f95709138a43b934997a15555d3e42655a02bee3",
            "comment": "feat(eslint-config): Reconfigure TS unused vars"
          }
        ]
      }
    },
    {
      "date": "Wed, 15 May 2024 06:47:18 GMT",
      "version": "0.3.3",
      "tag": "@langri-sha/babel-test_v0.3.3",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "92c7d4c9ac50342a39af3fc029307a2899f0ae34",
            "comment": "chore(deps): chore(deps): update dependency @types/node to v20.12.12"
          }
        ]
      }
    },
    {
      "date": "Tue, 14 May 2024 20:41:06 GMT",
      "version": "0.3.2",
      "tag": "@langri-sha/babel-test_v0.3.2",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "bb5eca0acea87c11cf873197bea7383eef3f8465",
            "comment": "chore(deps): fix(deps): update dependency ramda to v0.30.0"
          },
          {
            "author": "email not defined",
            "package": "@langri-sha/babel-test",
            "commit": "bb5eca0acea87c11cf873197bea7383eef3f8465",
            "comment": "chore(deps): chore(deps): update node.js to v20.13.1"
          }
        ]
      }
    },
    {
      "date": "Fri, 03 May 2024 12:54:50 GMT",
      "version": "0.3.1",
      "tag": "@langri-sha/babel-test_v0.3.1",
      "comments": {
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "a2e01a6b6e6b2243be9c5cdd2e77fe7cc162b6de",
            "comment": "fix(workspace): Correct published paths"
          }
        ]
      }
    },
    {
      "date": "Fri, 03 May 2024 10:44:32 GMT",
      "version": "0.3.0",
      "tag": "@langri-sha/babel-test_v0.3.0",
      "comments": {
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "68e384490ef3496631d9e74f059a50a466179b5d",
            "comment": "fix(workspace): Revise build command"
          }
        ],
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "976a3bb428390ade32e6a6a4332d335d4a39915b",
            "comment": "feat(project): Configure package build"
          }
        ]
      }
    },
    {
      "date": "Wed, 24 Apr 2024 22:03:36 GMT",
      "version": "0.2.0",
      "tag": "@langri-sha/babel-test_v0.2.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "f9dcfcb1ce5b663131c076f5364425c03f757710",
            "comment": "feat(workspace): Update package metadata"
          }
        ],
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/babel-test",
            "commit": "8d67a370e43a304d96b4e84a0eb773a9f7d4fecd",
            "comment": "feat(workspace): Use PNPM local references"
          }
        ]
      }
    }
  ]
}
