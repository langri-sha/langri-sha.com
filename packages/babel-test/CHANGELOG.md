# Change Log - @langri-sha/babel-test

This log was last generated on Fri, 26 Jul 2024 21:00:29 GMT and should not be manually modified.

<!-- Start content -->

## 0.6.2

Fri, 26 Jul 2024 21:00:29 GMT

### Patches

- chore(deps): chore(deps): update dependency @types/node to v20.14.12 (email not defined)

## 0.6.1

Wed, 17 Jul 2024 10:48:16 GMT

### Patches

- fix(workspace): Use correct GitHub directory (filip.dupanovic@gmail.com)
- chore(deps): chore(deps): update dependency @types/node to v20.14.11 (email not defined)

## 0.6.0

Mon, 08 Jul 2024 21:43:05 GMT

### Minor changes

- feat(workspace): Use `vitest` (filip.dupanovic@gmail.com)

### Patches

- chore(deps): chore(deps): update dependency @types/node to v20.14.5 (email not defined)
- chore(deps): chore(deps): update node.js to v20.15.0 (email not defined)
- fix(projen): Use correct output directory (filip.dupanovic@gmail.com)
- chore(deps): chore(deps): update dependency @types/node to v20.14.6 (email not defined)
- chore(deps): chore(deps): update dependency @types/ramda to v0.30.1 (email not defined)
- chore(deps): chore(deps): update dependency @types/node to v20.14.3 (email not defined)
- chore(deps): chore(deps): update dependency @types/node to v20.14.10 (email not defined)
- chore(deps): chore(deps): update dependency @types/node to v20.14.7 (email not defined)
- chore(deps): chore(deps): update dependency @types/node to v20.14.8 (email not defined)

## 0.5.0

Sat, 15 Jun 2024 09:09:06 GMT

### Minor changes

- feat(babel-preset): Hoist TS types to peers (filip.dupanovic@gmail.com)
- feat(babel-test): Manage package with projen (filip.dupanovic@gmail.com)
- feat(projen): Reconfigure project expressions (filip.dupanovic@gmail.com)

### Patches

- chore(deps): chore(deps): update node.js to v20.14.0 (email not defined)
- chore(deps): chore(deps): update dependency @types/node to v20.14.2 (email not defined)
- chore(deps): fix(deps): update dependency ramda to v0.30.1 (email not defined)

## 0.4.0

Tue, 21 May 2024 05:22:50 GMT

### Minor changes

- feat(eslint-config): Reconfigure TS unused vars (filip.dupanovic@gmail.com)

## 0.3.3

Wed, 15 May 2024 06:47:18 GMT

### Patches

- chore(deps): chore(deps): update dependency @types/node to v20.12.12 (email not defined)

## 0.3.2

Tue, 14 May 2024 20:41:06 GMT

### Patches

- chore(deps): fix(deps): update dependency ramda to v0.30.0 (email not defined)
- chore(deps): chore(deps): update node.js to v20.13.1 (email not defined)

## 0.3.1

Fri, 03 May 2024 12:54:50 GMT

### Patches

- fix(workspace): Correct published paths (filip.dupanovic@gmail.com)

## 0.3.0

Fri, 03 May 2024 10:44:32 GMT

### Minor changes

- feat(project): Configure package build (filip.dupanovic@gmail.com)

### Patches

- fix(workspace): Revise build command (filip.dupanovic@gmail.com)

## 0.2.0

Wed, 24 Apr 2024 22:03:36 GMT

### Minor changes

- feat(workspace): Update package metadata (filip.dupanovic@gmail.com)

### Patches

- feat(workspace): Use PNPM local references (filip.dupanovic@gmail.com)
