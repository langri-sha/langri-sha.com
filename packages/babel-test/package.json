{
  "name": "@langri-sha/babel-test",
  "version": "0.6.2",
  "bugs": {
    "url": "https://github.com/langri-sha/langri-sha.com/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/langri-sha/langri-sha.com.git",
    "directory": "packages/babel-test"
  },
  "license": "MIT",
  "author": {
    "name": "Filip DupanoviÄ‡",
    "email": "filip.dupanovic@gmail.com",
    "url": "https://langri-sha.com",
    "organization": false
  },
  "type": "module",
  "main": "src/index.ts",
  "scripts": {
    "prepublishOnly": "rm -rf dist; tsc --project tsconfig.build.json"
  },
  "dependencies": {
    "ramda": "0.30.1"
  },
  "devDependencies": {
    "@langri-sha/monorepo": "workspace:*",
    "@langri-sha/tsconfig": "workspace:*",
    "@langri-sha/vitest": "workspace:*",
    "@types/babel__core": "^7.8.0",
    "@types/node": "20.14.12",
    "@types/ramda": "0.30.1"
  },
  "peerDependencies": {
    "@babel/core": "^7.8.0"
  },
  "publishConfig": {
    "access": "public",
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.ts and run \"npx projen\"."
}
