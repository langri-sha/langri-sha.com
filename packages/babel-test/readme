# @langri-sha/babel-preset

This is Babel preset that suitable for targetting modern browsers and runtimes,
with support for [TypeScript], [Emotion] and [React].

## Usage

Install the preset and it's realted dependencies:

```sh
npm install -D @babel/core @langri-sha/babel-preset
```

Add a root Babel configuration:

```js
// babel.config.js
module.exports = {
  presets: ['@langri-sha/babel-preset'],
}
```

For usage with [Webpack] and [`babel-loader`]:

```js
// webpack.config.js
module.exports = {
  module: {
    rules: [
      {
        test: /\.(c|m)?(t|j)sx?$/,
        loader: 'babel-loader',
      },
    ],
  },
}
```

## See

- [Using Babel]

[`babel-loader`]: https://www.npmjs.com/package/babel-loader
[emotion]: https://emotion.sh/docs/babel
[react]: https://react.dev/learn
[typescript]:
  https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html
[using babel]: https://babeljs.io/setup
[webpack]: https://webpack.js.org/loaders/babel-loader/
