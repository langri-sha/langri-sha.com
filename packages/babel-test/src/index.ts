import type { PluginItem, TransformOptions } from '@babel/core'
import * as babel from '@babel/core'
import monorepo from '@langri-sha/monorepo'
import * as R from 'ramda'

export type Preset = {
  plugins: PluginItem[]
  presets: PluginItem[]
}

/*
 * For a given preset, returns the list of loaded plugins, transformed in a way
 * that they can be serialized across different environments.
 */
export const loadPresetPlugins = async (
  envName: string,
  preset: PluginItem,
): Promise<Array<[name: string, options: Record<string, unknown>]>> => {
  // @ts-expect-error: Missing `babel.loadOptionsAsync`.
  const { plugins } = await babel.loadOptionsAsync(options(envName, preset))

  // @ts-expect-error: `any[][]` is not assignable to returned tuple.
  return R.pipe(
    R.map(transformPaths),
    R.map(R.map(R.when(R.is(Object), transformNodeVersion))),
  )(plugins)
}

const options = (envName: string, preset: PluginItem): TransformOptions => ({
  babelrc: false,
  configFile: false,
  filename: module?.parent?.filename,
  envName,
  presets: [preset],
})

const transformPaths = ({
  key,
  options,
}: {
  key: string
  options: Record<string, unknown>
}): [string, Record<string, unknown>] => [
  key.replace(monorepo.root, '<WORKSPACE>'),
  options,
]

const transformNodeVersion = R.evolve({
  targets: R.evolve({
    node: R.when(
      R.equals(process.version.slice(1)),
      R.always('%NODE_CURRENT%'),
    ),
  }),
})
