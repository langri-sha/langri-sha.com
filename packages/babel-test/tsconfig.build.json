{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@langri-sha/tsconfig/build.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib"
  }
}
