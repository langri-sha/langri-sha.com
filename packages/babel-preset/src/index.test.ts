import { test, expect } from '@langri-sha/jest-test'

import { loadPresetPlugins } from '@langri-sha/babel-test'

test('default preset plugins', async () => {
  for (const environment of ['development', 'production']) {
    const plugins = await loadPresetPlugins(
      environment,
      '@langri-sha/babel-preset',
    )

    expect(plugins).toMatchSnapshot(`${environment} environment`)
  }
})
