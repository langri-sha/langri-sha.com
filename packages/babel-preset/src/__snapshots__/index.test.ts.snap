// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default preset plugins: development environment 1`] = `
[
  [
    "proposal-export-default-from",
    {},
  ],
  [
    "@emotion",
    {},
  ],
  [
    "transform-typescript",
    {
      "allowDeclareFields": undefined,
      "allowNamespaces": true,
      "disallowAmbiguousJSXLike": false,
      "isTSX": false,
      "jsxPragma": undefined,
      "jsxPragmaFrag": "React.Fragment",
      "onlyRemoveTypeImports": undefined,
      "optimizeConstEnums": false,
    },
  ],
  [
    "transform-react-jsx",
    {
      "importSource": undefined,
      "pragma": "React.createElement",
      "pragmaFrag": "React.Fragment",
      "pure": undefined,
      "runtime": "classic",
      "throwIfNamespace": true,
      "useBuiltIns": false,
      "useSpread": undefined,
    },
  ],
  [
    "transform-react-display-name",
    {},
  ],
  [
    "transform-react-pure-annotations",
    {},
  ],
  [
    "transform-flow-strip-types",
    {
      "all": undefined,
      "allowDeclareFields": undefined,
    },
  ],
  [
    "transform-unicode-sets-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-class-static-block",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-private-property-in-object",
    {
      "loose": "#__internal__@babel/preset-env__prefer-false-but-true-is-ok-if-it-prevents-an-error",
    },
  ],
  [
    "transform-class-properties",
    {
      "loose": "#__internal__@babel/preset-env__prefer-false-but-true-is-ok-if-it-prevents-an-error",
    },
  ],
  [
    "transform-private-methods",
    {
      "loose": "#__internal__@babel/preset-env__prefer-false-but-true-is-ok-if-it-prevents-an-error",
    },
  ],
  [
    "transform-numeric-separator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-logical-assignment-operators",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-nullish-coalescing-operator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-optional-chaining",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-json-strings",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-optional-catch-binding",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-parameters",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-async-generator-functions",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-object-rest-spread",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-dotall-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-unicode-property-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-named-capturing-groups-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-async-to-generator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-exponentiation-operator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-template-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-function-name",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-arrow-functions",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-block-scoped-functions",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-classes",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-object-super",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-shorthand-properties",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-duplicate-keys",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-computed-properties",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-for-of",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-sticky-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-unicode-escapes",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-unicode-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-spread",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-destructuring",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-block-scoping",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-typeof-symbol",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-new-target",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-regenerator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-member-expression-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-property-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-reserved-words",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-export-namespace-from",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-modules-commonjs",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-dynamic-import",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "syntax-top-level-await",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "syntax-import-meta",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
]
`;

exports[`default preset plugins: production environment 1`] = `
[
  [
    "proposal-export-default-from",
    {},
  ],
  [
    "@emotion",
    {},
  ],
  [
    "transform-typescript",
    {
      "allowDeclareFields": undefined,
      "allowNamespaces": true,
      "disallowAmbiguousJSXLike": false,
      "isTSX": false,
      "jsxPragma": undefined,
      "jsxPragmaFrag": "React.Fragment",
      "onlyRemoveTypeImports": undefined,
      "optimizeConstEnums": false,
    },
  ],
  [
    "transform-react-jsx",
    {
      "importSource": undefined,
      "pragma": "React.createElement",
      "pragmaFrag": "React.Fragment",
      "pure": undefined,
      "runtime": "classic",
      "throwIfNamespace": true,
      "useBuiltIns": false,
      "useSpread": undefined,
    },
  ],
  [
    "transform-react-display-name",
    {},
  ],
  [
    "transform-react-pure-annotations",
    {},
  ],
  [
    "transform-flow-strip-types",
    {
      "all": undefined,
      "allowDeclareFields": undefined,
    },
  ],
  [
    "transform-unicode-sets-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-class-static-block",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-private-property-in-object",
    {
      "loose": "#__internal__@babel/preset-env__prefer-false-but-true-is-ok-if-it-prevents-an-error",
    },
  ],
  [
    "transform-class-properties",
    {
      "loose": "#__internal__@babel/preset-env__prefer-false-but-true-is-ok-if-it-prevents-an-error",
    },
  ],
  [
    "transform-private-methods",
    {
      "loose": "#__internal__@babel/preset-env__prefer-false-but-true-is-ok-if-it-prevents-an-error",
    },
  ],
  [
    "transform-numeric-separator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-logical-assignment-operators",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-nullish-coalescing-operator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-optional-chaining",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-json-strings",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-optional-catch-binding",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-parameters",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-async-generator-functions",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-object-rest-spread",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-dotall-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-unicode-property-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-named-capturing-groups-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-async-to-generator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-exponentiation-operator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-template-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-function-name",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-arrow-functions",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-block-scoped-functions",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-classes",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-object-super",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-shorthand-properties",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-duplicate-keys",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-computed-properties",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-for-of",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-sticky-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-unicode-escapes",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-unicode-regex",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-spread",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-destructuring",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-block-scoping",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-typeof-symbol",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-new-target",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-regenerator",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-member-expression-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-property-literals",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-reserved-words",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-export-namespace-from",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-modules-commonjs",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "transform-dynamic-import",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "syntax-top-level-await",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
  [
    "syntax-import-meta",
    {
      "loose": undefined,
      "spec": false,
      "useBuiltIns": false,
    },
  ],
]
`;
